examples/offline_inference/chat_with_tools.py

```diff
@@ -120,21 +125,33 @@ def get_current_weather(city: str, state: str, unit: 'str'):
 
 # let's now actually parse and execute the model's output simulating an API call by using the
 # above defined function
-tool_calls = json.loads(output)
-tool_answers = [
-    tool_funtions[call['name']](**call['arguments']) for call in tool_calls
-]
-
-# append the answer as a tool message and let the LLM give you an answer
-messages.append({
-    "role": "tool",
-    "content": "\n\n".join(tool_answers),
-    "tool_call_id": generate_random_id(),
-})
+# Parse the output as JSON
+tool_calls = None
+try:
+    tool_calls = json.loads(output)
+    tool_answers = [
+        tool_funtions[call['name']](**call['arguments']) for call in tool_calls
+    ]
+
+    # append the answer as a tool message and let the LLM give you an answer
+    messages.append({
+        "role": "tool",
+        "content": "\n\n".join(tool_answers),
+        "tool_call_id": generate_random_id(),
+    })
+except json.JSONDecodeError:
+    print(
+        "\n\033[91mJSON parsing failed, the model returned non-structured content:\033[0m\n",
```

Comment:
We already have the `rich` library in the test dependencies. Perhaps it would be better to use that to output colored text because hopefully it will detect whether stdout can handle colors or not.

Reply:
thank for you feedback. based on the other examples without color, so remove it.